extension/chrome/elements/compose-modules/compose-send-btn-module.ts

```diff
@@ -197,21 +199,26 @@ export class ComposeSendBtnModule extends ViewModule<ComposeView> {
   };
 
 
-  private doSendMsg = async (msg: SendableMsg) => {
+  private doSendMsgs = async (msgs: SendableMsg[]) => {
     // if this is a password-encrypted message, then we've already shown progress for uploading to backend
     // and this requests represents second half of uploadable effort. Else this represents all (no previous heavy requests)
     const progressRepresents = this.view.pwdOrPubkeyContainerModule.isVisible() ? 'SECOND-HALF' : 'EVERYTHING';
     let msgSentRes: GmailRes.GmailMsgSend;
-    try {
-      this.isSendMessageInProgress = true;
-      msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-    } catch (e) {
-      if (msg.thread && ApiErr.isNotFound(e) && this.view.threadId) { // cannot send msg because threadId not found - eg user since deleted it
-        msg.thread = undefined;
+    for (const msg of msgs) {
+      try {
+        this.isSendMessageInProgress = true;
         msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-      } else {
-        this.isSendMessageInProgress = false;
-        throw e;
+      } catch (e) {
+        if (msg.thread && ApiErr.isNotFound(e) && this.view.threadId) { // cannot send msg because threadId not found - eg user since deleted it
+          msg.thread = undefined;
+          msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
+        } else {
+          this.isSendMessageInProgress = false;
```

Comment:
should investigate what is `isSendMessageInProgress` used for. Maybe there should be one before the whole loop and one after.

Error handling will be a real pain, too. what to do if a few messages were sent and then there was an error. At a minimum, we'll have to communicate it to the user such as `Messages to some recipients were sent successfully, while messages to others encountered an error: ...`

Reply:
ping @tomholub

Reply:
looks good

>  Also a question -- how to display non-critical errors (binding gateway ids of the sent messages) -- I called them supplementaryOperationError

You could toast the error using `Ui.toast`. I think that would be ok enough.

Reply:
Unsure. I'll let you decide, you can do whatever is simpler.